packages/rest/src/route.ts

```diff
@@ -296,12 +304,13 @@ function generatePathFromParameters(
 
 function getPathForOperation(
   program: Program,
+  diagnostics: Diagnostic[],
   operation: OperationType,
   routeFragments: string[],
   options: RouteOptions
 ): { path: string; pathFragment?: string; parameters: HttpOperationParameters } {
-  const parameters: HttpOperationParameters = getOperationParameters(program, operation);
-
+  const [parameters, parmetersDiagnostics] = getOperationParameters(program, operation);
+  diagnostics.push(...parmetersDiagnostics);
```

Comment:
wondering here if maybe instead of being an array or diagnositcs it could be a new type

```ts
interface DiagnositcCollector {
  add(diagnostic: Diagnostic);
  pipe<T>(result: [T, Diagnostics[]]): T;
}
```

which lets the following be used like
```ts
const parameters = diagnostics.pipe(getOperationParameters(program, operation));
```

Reply:
thinking this could just be let the the internals of the accessor meaning at the end you can have
```ts
return [myresult, diagnostics.diagnostics]
```